apps/protocol-frontend/src/components/DashboardShell.tsx

```diff
@@ -46,76 +48,47 @@ const DashboardShell = ({ user }: DashboardShellProps) => {
-  ) => {
-    if (selectedDaos) {
-      const contributionsCountResponse = await getUserContributionsCount(
-        subWeeks(new Date(), dateRangeValue),
-        new Date(),
-        selectedDaos.map(dao => dao.value).filter(dao => dao !== null),
-      );
-      setFunc(contributionsCountResponse);
-      if (selectedDaos.some(dao => dao.label === UNASSIGNED)) {
-        setFunc(contributionsCountResponse);
-      } else {
-        setFunc(
-          contributionsCountResponse?.filter(
-            contribution => contribution?.name !== UNASSIGNED,
-          ),
-        );
-      }
-      if (
-        selectedDaos.length === 0 &&
-        !selectedDaos.some(dao => dao.label === UNASSIGNED)
-      ) {
-        const contributionsCountResponse = await getUserContributionsCount(
-          subWeeks(new Date(), dateRangeValue),
-          new Date(),
-          combinedDaoListOptions
-            .map(dao => dao.value)
-            .filter(dao => dao !== null),
-        );
-        setFunc(contributionsCountResponse);
-      }
-    }
-  };
-
-  useEffect(() => {
-    fetchHeatMapCount(dateRange.value, setContributionsCount);
-  }, [user, dateRange, selectedDaos]);
-
-  useEffect(() => {
-    fetchHeatMapCount(52, setFullContributionsCount);
-  }, [user, selectedDaos]);
-
   const userDaoListOptions =
     userDaosListData?.map(dao => ({
       value: dao.id,
       label: dao.name ?? '',
     })) || [];
 
-  const unassignedContributions = [
-    {
-      value: Number(null),
-      label: UNASSIGNED,
-    },
-  ];
-
-  const dateRangeOptions = [
-    { value: 1, label: 'Last Week' },
-    { value: 4, label: 'Last Month' },
-    { value: 12, label: 'Last Quarter' },
-    { value: 52, label: 'Last Year' },
-  ];
-
   const combinedDaoListOptions = [
     ...new Set([...unassignedContributions, ...userDaoListOptions]),
   ];
 
+  const { data: fullContributionsCount } = useContributionCountInYear({
+    endDate: endOfDay(TODAY_DATE),
+    guildIds:
+      selectedDaos.length === 0 &&
+      !selectedDaos.some(dao => dao.label === UNASSIGNED)
+        ? combinedDaoListOptions
+            .map(dao => dao.value)
+            .filter(dao => dao !== null)
+        : selectedDaos.map(dao => dao.value).filter(dao => dao !== null),
+    excludeUnassigned: !(
+      selectedDaos.length === 0 ||
+      selectedDaos.some(dao => dao.label === UNASSIGNED)
+    ),
+  });
+
+  const { data: contributionsCount } = useContributionCountInRange({
+    startDate: subWeeks(startOfDay(TODAY_DATE), dateRange.value),
+    endDate: endOfDay(TODAY_DATE),
+    guildIds:
+      selectedDaos.length === 0 &&
+      !selectedDaos.some(dao => dao.label === UNASSIGNED)
+        ? combinedDaoListOptions
+            .map(dao => dao.value)
+            .filter(dao => dao !== null)
+        : selectedDaos.map(dao => dao.value).filter(dao => dao !== null),
+    excludeUnassigned: !(
+      selectedDaos.length === 0 ||
+      selectedDaos.some(dao => dao.label === UNASSIGNED)
```

Comment:
Some of these can moved to variables and reused rather than recalculating the values in each argument. An example would be make `const unassignedSelected = selectedDaos.some(dao => dao.label === UNASSIGNED)`. Rather than recalculating in each argument.